pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -132,6 +132,15 @@ func (ssc *defaultStatefulSetControl) UpdateStatefulSet(set *apps.StatefulSet, p
 		if !isCreated(replicas[i]) {
 			return ssc.podControl.CreateStatefulPod(set, replicas[i])
 		}
+
+		// If we find a Pod that is currently terminating, we must wait until graceful deletion
+		// completes before we continue to make  progress.
+		if isTerminated(replicas[i]) {
```

Comment:
Should this helper be named `isTerminating` since the pod is not terminated when it has a deletion timestamp?